### DISCUSSION TITLE
How to specify executables --build-id  ( .note.gnu.build-id in elf) ?

### DISCUSSION THREAD

**Discussion by zckv** [pyinstaller/pyinstaller#7934] [Help] [answered]
Hello 👋

I searched issue and discussions, but nobody seems to have this problem. I may be misusing the tool, I'm really sorry if it's the case.

When building with pyinstaller, the executable always have the same build-id ( .note.gnu.build-id in elf):

```sh
pyinstaller hello.py --onefile
readelf -n dist/hello

  Owner                Data size        Description
  GNU                  0x00000014       NT_GNU_BUILD_ID (unique build ID bitstring)
    Build ID: 3dfe013b3027047470d1aac10a3504baf6969725
```

I am a beginner, but I guess that it should be different for each application. It is the same in onefolder also. 
It seems that there is a linker option to specify this ("--build-id"), but it is not in the source code.

**Comment by rokm** [comment, marked as answer]
PyInstaller uses pre-compiled bootloader executable, so all frozen applications made with that bootloader executable will have the same `NT_GNU_BUILD_ID`. If you wanted it to be different for each application build, you would need to recompile the bootloader prior to building the application. Or find and edit the value in the generated executable.
